src/platform/forms-system/src/js/review/ObjectField.jsx

```diff
@@ -118,8 +118,14 @@ class ObjectField extends React.Component {
         });
         if (objectFields.length > 1 && visible.length > 0) {
           return objectFields
-            .filter(() =>
-              showReviewField(first, schema, uiSchema, formData, formContext),
+            .filter(propName =>
+              showReviewField(
+                propName,
```

Comment:
This is the main change. The bug was that we were choosing for each field whether to `showReviewField` based only on the top-level field. The fix is to determine whether to `showReviewField` for each field individually.